### PR TITLE
RT-12227: Implemented payProduct method; Added OrderResourceResponse;…

### PR DESCRIPTION
**Pull request by yurez**
… Fixed Serializer for empty collection responses; Added tests; Added documentation to README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,109 @@
-# levelcredit-api-php
+# LevelCredit API Client
+
+LevelCredit API Client is part of LevelCredit SDKs. The Client was created like wrapper for send requests to API.
```
Comment (caryp) [inline README.md:3]: LevelCredit API Client is low-level dependency of the LevelCredit SDK libraries.  This project is alpha stage with no guarantee with backward compatibility in future versions.  It is not recommended to program directly to this project, but instead, use the public methods to our higher-level SDK libraries. We reserve the right to remove this project completely.  Consider yourself warned.

File: README.md [hunk 2]
```diff
@@ -1 +1,109 @@
-# levelcredit-api-php
+# LevelCredit API Client
+
+LevelCredit API Client is part of LevelCredit SDKs. The Client was created like wrapper for send requests to API. 
+The client use [Guzzle](https://github.com/guzzle/guzzle) for send http requests.
```
Comment (caryp) [inline README.md:4]: change `use` to `uses` 
change `for` to `to`

File: README.md [hunk 3]
```diff
@@ -1 +1,109 @@
-# levelcredit-api-php
+# LevelCredit API Client
+
+LevelCredit API Client is part of LevelCredit SDKs. The Client was created like wrapper for send requests to API. 
+The client use [Guzzle](https://github.com/guzzle/guzzle) for send http requests. 
+The client supports adding log handler for logging messages [LogHandlerInterface](https://github.com/levelcredit/levelcredit-api-php/blob/master/src/Logging/LogHandlerInterface.php).
```
Comment (caryp) [inline README.md:5]: The client supports adding a log handler for logging messages. See [LogHandlerInterface](https://github.com/levelcredit/levelcredit-api-php/blob/master/src/Logging/LogHandlerInterface.php).

File: README.md [hunk 4]
```diff
@@ -1 +1,109 @@
...
+```php
+<?php
+
+use LevelCredit\LevelCreditApi\LevelCreditApiClient;
+use LevelCredit\LevelCreditApi\Exception\LevelCreditApiException;
+use LevelCredit\LevelCreditApi\Enum\UserEmbeddedEntities;
+use LevelCredit\LevelCreditApi\Model\Request\GetPartnerUsersFilter;
+use LevelCredit\LevelCreditApi\Model\Response\Resource\User;
+
+$client = LevelCreditApiClient::create('<CLIENT_ID>', '<CLIENT_SECRET>') // client_id and client_secret are optional
```
Comment (caryp) [inline README.md:24]: If you use env vars (like `ENV('CLIENT_ID')` ) then this code could be cut and pasted as is and run!

File: README.md [hunk 5]
```diff
@@ -1 +1,109 @@
...
+//    ->setClientId('<YOUR_CLIENT_ID>') // you can change client_id for each request if needed
+//    ->setClientSecret('<YOUR_CLIENT_SECRET>') // client_secret the same
+//    ->setBaseUri('<YOUR_SPECIFIC_BASE_URL>') // by default will use production levelcredit url
```
Comment (caryp) [inline README.md:27]: default to `https://my.sandbox2.renttrack.com` in this example.

File: README.md [hunk 6]
```diff
@@ -1 +1,109 @@
...
+    /** @var User $user */
+    $user = $usersCollection->getElements()->first();
+} catch (LevelCreditApiException $e) {
+    // process exception here can be guzzle (like problem with connection or certificate), logic or serialization errors
+}
+```
+
+## About
+
+### Components
```
Comment (caryp) [inline README.md:76]: remove unused sections

**Review by caryp — CHANGES_REQUESTED** [review]
Looking good.  just some minor copy changes.
Please also change version number to "0.0.1" since this is an alpha version.

**Review by tatyana-shmatchenko — APPROVED** [review]
Nice work :+1:

**Review by tatyana-shmatchenko — APPROVED** [review]

**Pull request merged**